subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -599,7 +599,8 @@ static int gatt_proxy_advertise(struct bt_mesh_subnet *sub)
 		}
 	}
 
-	if (sub->node_id == BT_MESH_NODE_IDENTITY_STOPPED) {
+	if (sub->node_id == BT_MESH_NODE_IDENTITY_STOPPED &&
```

Comment:
Previously, the advertising was always started here either with node or network identity advertising, `bt_mesh_adv_start` was always called and `adv_timeout` was always updated. Now, if node id is stopped and gatt proxy is disabled, `bt_mesh_adv_start` won't be called and `adv_timeout` will be `SYS_FOREVER_MS`. When the node is in one subnet only, this is fine. But if there are two or more subnets, the `remaining` time will be reduced, but `adv_timeout` will stay unchanged, which will block adv thread until either `bt_mesh_adv_update` is called or someone pushes a packet into adv queue.

Reply:
Hm. Seems you are right. Needs loop through subnetworks here.